builtin/plugin/v5/backend.go

```diff
@@ -72,6 +73,12 @@ func (b *backend) reloadBackend(ctx context.Context) error {
 	}
 	b.Backend = nb
 
+	// Re-initialize the backend in case plugin was reloaded
+	// after it crashed
+	b.Backend.Initialize(ctx, &logical.InitializationRequest{
```

Comment:
Would this be an issue if Initialize was called once again outside of this? According to https://github.com/hashicorp/vault/pull/14946/files#r953130550, we may call this in mount.go or auth.go, so an explicit reload would be invoking `Initialize` twice now IIUC.

Reply:
I'm seeing that `InitializeFunc` isn't called after `vault plugin reload` on both an intentionally killed plugin process and a running plugin process. I then expected that sending a request to the plugin _after_ the successful reload (i.e., non-initial lazy loading) would result in `InitializeFunc` being called, but it doesn't. 

I still think we need to do what @vinay-gopalan has done in this PR to initialize properly after a plugin process is killed. The reload scenario described above could be treated as a separate issue.

Reply:
+1, sorry for the delay in the response, but I also am noticing that the `InitializeFunc` is only invoked for a plugin process (after it is killed) if we add these updates  to `reloadBackend`. Since after killing a process we aren't going through the `mount` function in `mount.go`, the `InitializeFunc` is not invoked.